src/hotCodeReplace.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+
+const runningSessions: Set<vscode.DebugSession> = new Set();

```

Comment:
Find no one is using `runningSessions`. What's the purpose of this cache?

Reply:
Some code is removed in this review for custom events.